### PR TITLE
Update metasearch button style and use a modal for screens below `lg`

### LINKED ISSUE #2668: New Metasearch button
## Description

As part of the design changes in #460, we need to add the metasearch button that triggers a popover in a modal. The component has two sizes and follows the breakpoint changes of "Load more" button.

image

## References
* [Figma component](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5327-30212&mode=dev) in Design Library.

### PR DESCRIPTION
**Pull request by chenikabukes**
## Fixes

Fixes #2668 by @fcoveram 

## Description

This PR addressed the design improvement in #460 by making the external sources popup a modal and adjusting the modal title according to screen size.

## Testing Instructions

1. Conduct a search
2. Scroll to the bottom to find the new external search button
3. Click on the button to let the modal display
4. Change screen size to view the different modal titles according to screen size

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by chenikabukes** [comment]
*Note PR is addressing the same issue @obulat, but I was encountering build errors when I merged in the latest main and rather started fresh :). This PR addresses the comments on the previous PR.

File: frontend/src/components/VExternalSearch/VExternalSearchForm.vue [hunk 1]
```diff
@@ -5,62 +5,68 @@
...
       id="external-sources-button"
       ref="triggerRef"
       :pressed="triggerA11yProps['aria-expanded']"
       aria-haspopup="dialog"
-      :aria-controls="
-        isMd ? 'external-sources-popover' : 'external-sources-modal'
-      "
-      variant="dropdown-label"
+      :aria-controls="'external-sources-modal'"
+      variant="filled-gray"
```
Comment (obulat) [inline frontend/src/components/VExternalSearch/VExternalSearchForm.vue:14]:
```suggestion
      variant="bordered-gray"
```

File: frontend/src/components/VExternalSearch/VExternalSearchForm.vue [hunk 2]
```diff
@@ -5,62 +5,68 @@
...
+        v-else
+        path="externalSources.form.noResultsTitle"
+        tag="p"
+        class="description-regular"
+      >
+        <template #type>{{
+          $t(`externalSources.form.types.${type}`)
+        }}</template>
+        <template #query>{{ searchTerm }}</template>
+      </i18n>
```
Comment (obulat) [inline frontend/src/components/VExternalSearch/VExternalSearchForm.vue:55]:
```suggestion
```

File: frontend/src/components/VExternalSearch/VExternalSearchForm.vue [hunk 3]
```diff
@@ -5,62 +5,68 @@
...
+      variant="filled-gray"
       size="disabled"
-      class="caption-regular ms-2 min-w-max gap-1 px-3 py-1 pe-1 text-dark-charcoal focus-visible:border-tx"
+      class="label-bold lg:description-bold h-16 w-full lg:h-18"
       @click="onTriggerClick"
-      >{{ $t("externalSources.button").toString()
-      }}<VIcon
+    >
+      <i18n
+        v-if="!hasNoResults && isSupported && isMd"
```
Comment (obulat) [inline frontend/src/components/VExternalSearch/VExternalSearchForm.vue:20]:
```suggestion
        v-if="isMd"
```

File: frontend/src/components/VExternalSearch/VExternalSearchForm.vue [hunk 4]
```diff
@@ -5,62 +5,68 @@
...
+        path="externalSources.form.supportedTitle"
+        tag="p"
+        class="description-regular"
+      />
+
+      <i18n
+        v-else-if="!hasNoResults && isSupported && !isMd"
```
Comment (obulat) [inline frontend/src/components/VExternalSearch/VExternalSearchForm.vue:27]:
```suggestion
        v-else
```

File: frontend/src/components/VExternalSearch/VExternalSearchForm.vue [hunk 5]
```diff
@@ -5,62 +5,68 @@
...
+        tag="p"
+        class="description-regular"
+      >
+        <template #type>{{
+          $t(`externalSources.form.types.${type}`)
+        }}</template>
+        <template #query>{{ searchTerm }}</template>
+      </i18n>
+      <VIcon
         class="text-dark-charcoal-40"
```
Comment (obulat) [inline frontend/src/components/VExternalSearch/VExternalSearchForm.vue:57]:
```suggestion
```

File: frontend/src/components/VExternalSearch/VExternalSearchForm.vue [hunk 6]
```diff
@@ -5,62 +5,68 @@
...
         name="caret-down"
       />
     </VButton>
     <template v-if="triggerElement">
-      <Component
-        :is="isMd ? 'VPopoverContent' : 'VModalContent'"
-        :id="isMd ? 'external-sources-popover' : 'external-sources-modal'"
-        :aria-labelledby="'external-sources-button'"
+      <VModalContent
+        ref="modalContentRef"
```
Comment (obulat) [inline frontend/src/components/VExternalSearch/VExternalSearchForm.vue:64]:
```suggestion
```

**Comment by chenikabukes** [comment]
I have updated en.json with the new object, wondering how the other language files will be updated?

**Review by obulat — CHANGES_REQUESTED** [review]
The visual regression test snapshots need to be updated using `just frontend/run test:playwright -u` (with the -u flag that tells the test to update the snapshots instead of throwing an error. There is one error that needs to be fixed in the `e2e` tests, too.
One more thing is that the `pnpm-lock.yaml` file should not have any changes. Could you revert the changes to it?

**Comment by obulat** [comment]
@chenikabukes, I rebased your PR onto main because some related changes were merged. I also updated the `ar.json` file in tests, so now this PR needs to update the Playwright visual regression tests. Here is a patch that you could use to fix the test:
```
diff --git a/frontend/test/playwright/visual-regression/components/external-sources-section.spec.ts b/frontend/test/playwright/visual-regression/components/external-sources-section.spec.ts
--- a/frontend/test/playwright/visual-regression/components/external-sources-section.spec.ts	(revision d0f51eb0e2942580d00590a7197bdec00abe3448)
+++ b/frontend/test/playwright/visual-regression/components/external-sources-section.spec.ts	(date 1692169772790)
@@ -14,14 +14,16 @@
 
 for (const dir of languageDirections) {
   for (const mediaType of supportedMediaTypes) {
-    breakpoints.describeMobileAndDesktop(async ({ expectSnapshot }) => {
+    breakpoints.describeMobileAndDesktop(async ({ breakpoint, expectSnapshot }) => {
       test(`External ${mediaType} sources popover - ${dir}`, async ({
         page,
       }) => {
         await goToSearchTerm(page, "birds", { searchType: mediaType, dir })
 
+        const buttonTextKey = ["xs", "sm"].includes(breakpoint)
+          ? "externalSources.form.supportedTitleSM" : "externalSources.form.supportedTitle"
         const externalSourcesButton = page.getByRole("button", {
-          name: t("externalSources.button", dir),
+          name: t(buttonTextKey, dir),
         })
 ``` 
You would also need to update the snapshots after this by running `just frontend/run test:playwright -u`. Make sure that Docker is running on your computer when you do.

**Review by obulat — APPROVED** [review]
@chenikabukes, thank you for your contribution ❤️ 

I've rebased this PR onto main after the changes to the external sources modal were merged, and updated the external search button locator in the Playwright tests - and now the CI is green.

**Review by fcoveram — APPROVED** [review]
It looks great now ✨ 🚀

**Pull request merged**